nitrocli/src/commands.rs

```diff
@@ -171,41 +174,30 @@ where
   .map_err(|(_data, err)| err)
 }
 
-/// Pretty print the common status of all Nitrokey devices.
-fn print_status(
-  model: &'static str,
-  smartcard_id: &str,
-  firmware_major: i32,
-  firmware_minor: i32,
-  user_retries: u8,
-  admin_retries: u8,
-) {
+/// Query and pretty print the common status of all Nitrokey devices.
+fn print_status(model: &'static str, device: &nitrokey::DeviceWrapper) -> Result<()> {
+  let serial_number = device
+    .get_serial_number()
+    .map_err(|err| get_error("Could not query the serial number", &err))?;
   println!(
     r#"Status:
   model:             {model}
-  smart card ID:     {id}
+  smart card ID:     0x{id}
   firmware version:  {fwv0}.{fwv1}
   user retry count:  {urc}
   admin retry count: {arc}"#,
     model = model,
-    id = smartcard_id,
-    fwv0 = firmware_major,
-    fwv1 = firmware_minor,
-    urc = user_retries,
-    arc = admin_retries,
+    id = serial_number,
```

Comment:
Does it make sense to differentiate between the smart card ID (for the storage device) and the serial number for the Pro? The string `smart card ID` seems to be misleading now, and I am trying to understand whether to move this part into the device specific print function or keep it here and change the string to `serial number`.

Reply:
I’d prefer to have this as a common field.  It is the same ID that we will show in the pinentry dialog (see #24) and that is reported by `gpg --card-status` in the same way for both models.   I’ll rename it to *serial number*.

That said, for my Nitrokey Storage, the serial number returned by `NK_get_status_storage` and `NK_device_serial_number` do not match the number reported by `gpg --card-status`.  I’ll have to investigate that.  Could you please check the result of `NK_device_serial_number` for your Storage stick?

Reply:
Sorry, I did not get to that today. I will get back to you tomorrow.

Reply:
I opened Nitrokey/nitrokey-storage-firmware#76 for this issue.

Reply:
No, I’ve never experienced such behavior.  But I’ve never used GnuPG on my development stick, and I only occasionally run `nitrocli` with my production stick.  And I do not use the Storage volumes at all.  So there might be interaction problems that I did not trigger with my usage pattern.  (That reminds me that we have to implement a `--debug` option that enables the debugging output from `libnitrokey`.)